interactive_ai/workflows/otx_domain/trainer/otx_v2/Dockerfile.gpu

```diff
@@ -1,84 +1,68 @@
+    NVIDIA_DRIVER_CAPABILITIES=compute,utility \
+    NVIDIA_VISIBLE_DEVICES=all \
+    LD_LIBRARY_PATH=/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/usr/local/lib \
+    PATH=/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
+    TORCH_CUDA_ARCH_LIST="6.0;6.1;6.2;7.0;7.2;7.5;8.0;8.6" \
+    TORCH_DONT_CHECK_COMPILER_ABI=1 \
+    UV_COMPILE_BYTECODE=1 \
+    UV_LINK_MODE=copy \
+    UV_PYTHON_DOWNLOADS=0 \
+    HF_HUB_OFFLINE=1
+
+# Install CUDA dependencies
+RUN apt update && \
+    apt install -y wget --no-install-recommends && \
+    wget https://developer.download.nvidia.com/compute/cuda/repos/debian12/x86_64/cuda-keyring_1.1-1_all.deb && \
+    dpkg -i cuda-keyring_1.1-1_all.deb && \
+    rm -rf cuda-keyring_1.1-1_all.deb && \
+    apt update && \
+    apt install -y --no-install-recommends \
+        cuda-libraries-12-6 \
+        cuda-compiler-12-6 && \
+    apt clean all && \
+    rm -rf /var/cache/apt/* && \
+    echo "/usr/local/nvidia/lib" >> /etc/ld.so.conf.d/nvidia.conf && \
+    echo "/usr/local/nvidia/lib64" >> /etc/ld.so.conf.d/nvidia.conf
 
-RUN apt update \
-	&& apt install -y wget --no-install-recommends
-
-RUN wget https://developer.download.nvidia.com/compute/cuda/repos/debian12/x86_64/cuda-keyring_1.1-1_all.deb \
-    && dpkg -i cuda-keyring_1.1-1_all.deb \
-    && rm -rf cuda-keyring_1.1-1_all.deb
-
-RUN apt update \
-	&& apt install -y --no-install-recommends \
-    cuda-libraries-12-6 \
-    cuda-compiler-12-6 \
-    && apt clean all \
-    && rm -rf /var/cache/apt/* \
-	&& echo "/usr/local/nvidia/lib" >> /etc/ld.so.conf.d/nvidia.conf \
-	&& echo "/usr/local/nvidia/lib64" >> /etc/ld.so.conf.d/nvidia.conf
-
-# step 2: get python dependencies
-FROM base AS build
-
-ENV UV_COMPILE_BYTECODE=1
-ENV UV_LINK_MODE=copy
-
-# Disable Python downloads, because we want to use the system interpreter across both images.
-ENV UV_PYTHON_DOWNLOADS=0
+# Install runtime dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        libgl1=1.6.* \
+        libglib2.0-0=2.74.* \
+        curl && \
+    rm -rf /var/lib/apt/lists/* && \
+    useradd -l -u 10001 non-root && \
+    pip3 uninstall -y setuptools pip wheel && \
+    rm -rf /root/.cache/pip
 
-# Copy the service dependencies
+# Copy application files
 COPY --link --from=libs . libs
-
 WORKDIR /interactive_ai/workflows/otx_domain/trainer/otx_v2
-
-COPY --link --from=ghcr.io/astral-sh/uv:0.6.12@sha256:515b886e8eb99bcf9278776d8ea41eb4553a794195ef5803aa7ca6258653100d /uv /bin/uv
-
 COPY --link scripts/ scripts
 COPY --link run run
 COPY --link download_pretrained_weights.py download_pretrained_weights.py
+COPY --link uv.lock uv.lock
+COPY --link pyproject.toml pyproject.toml
```

Comment:
I would suggest reusing the old pattern where these files are mounted for the RUN command. This leads to a slightly smaller image and matches the pattern used throughout docker files in Geti